### ISSUE TITLE
use tf to optimize this neural-style network loss,but how to update the input image with detail?

### ISSUE THREAD

**Issue by eroszip** [eroszip/neural-style#1]
First,thanks for your sharing with neural-style,after look over your code, i found you just use   train_step = tf.train.AdamOptimizer(learning_rate).minimize(loss)    to update the input, but i want to know the details behind this tech, could you give me some guidance? what is more, i also find there are some difference between  the tech with keras by fchollet, grads = K.gradients(loss, combination_image).
